app/main.py

```diff
@@ -1 +1,63 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(
+            self,
+            name: str
+    ) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(
+            self,
+            name: str
+    ) -> None:
+        super().__init__(name)
+        self.skills = ["JavaScript", "HTML", "CSS"]
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(
+            self,
+            name: str
+    ) -> None:
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django"]
+
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(
+            self,
+            name: str
+    ) -> None:
+        super().__init__(name)
+        self.skills = ["Java", "Android studio"]
+
+    def create_smooth_mobile_app(self) -> str:
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(BackendDeveloper, FrontendDeveloper):
+    def __init__(
+            self,
+            name: str
+    ) -> None:
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django", "JavaScript", "HTML", "CSS"]
+
```

Comment:
Skills are inherited from parent classes, no need to set them again.

Reply:
![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/7d32b4d6-d24f-4dbd-b3c1-345394d8134a)
![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/c9caa59b-d9fc-480d-aa83-58a10e95f9f9)


![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/28fae2ad-bde6-4cee-8686-4c86df4e8b0f)
![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/047cd4d7-859e-44b1-b425-b1d233a1e2c1)


![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/1695a9f6-246d-414a-9867-9b3fab8dd4fb)


![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/1e663057-f364-49b3-9e1a-2bb4be3aca51)
![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/f71c4187-3ffc-4610-8c6f-b8e5eac709c4)


in this case i understand: MRO, inheritance of  BackendDeveloper ```__init__```

**but WHY!**
![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/e1cc3f70-f23d-498d-860c-b9b75c7d9209)
![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/ebf8d557-4178-40e8-a570-faf043af4bf0)
![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/b38f48d7-e642-4171-820a-6e30f94594d0)
![image](https://github.com/mate-academy/py-software-engineers/assets/143685044/7bb095bf-c6eb-4a93-8740-33ee2b241934)

**WHY ["JavaScript", "HTML", "CSS"] come first**

if both ```__init__``` methods of father classes launch **WHY** second ```super().__init__ ```  in FrontendDeveloper **doesn't erase** 
```python 
["Python", "SQL", "Django"]
```
**to**
```python
self.skills = []
```

Reply:
Come to Q&A with this question